fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAuthenticationFilter.java

```diff
@@ -16,49 +16,62 @@
 
 package com.netflix.spinnaker.fiat.shared;
 
+import com.netflix.spinnaker.fiat.model.SpinnakerAuthorities;
+import com.netflix.spinnaker.fiat.model.UserPermission;
 import com.netflix.spinnaker.security.AuthenticatedRequest;
 import java.io.IOException;
-import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
 import javax.servlet.*;
 import lombok.extern.slf4j.Slf4j;
-import lombok.val;
 import org.springframework.security.authentication.AnonymousAuthenticationToken;
 import org.springframework.security.core.Authentication;
-import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
 
 @Slf4j
 public class FiatAuthenticationFilter implements Filter {
 
   private final FiatStatus fiatStatus;
+  private final FiatPermissionEvaluator permissionEvaluator;
 
-  public FiatAuthenticationFilter(FiatStatus fiatStatus) {
+  public FiatAuthenticationFilter(
+      FiatStatus fiatStatus, FiatPermissionEvaluator permissionEvaluator) {
     this.fiatStatus = fiatStatus;
+    this.permissionEvaluator = permissionEvaluator;
   }
 
   @Override
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
       throws IOException, ServletException {
+    Authentication auth;
     if (!fiatStatus.isEnabled()) {
-      chain.doFilter(request, response);
-      return;
+      List<GrantedAuthority> authorities =
```

Comment:
I'm a little confused on why this is happening when fiat is disabled.

Reply:
I was originally thinking that making an anonymous admin the security context when fiat was disabled, but that can be handled elsewhere.